[iOS] TestWebKitAPI.AppHighlights.AppHighlightCreateAndRestoreAndScroll is timing out

pxlcoder opened the pull request:
#### 6080e1230cac782ea81952cfea23f846771b7c50
<pre>
[iOS] TestWebKitAPI.AppHighlights.AppHighlightCreateAndRestoreAndScroll is timing out
<a href="https://bugs.webkit.org/show_bug.cgi?id=285882">https://bugs.webkit.org/show_bug.cgi?id=285882</a>
<a href="https://rdar.apple.com/141462057">rdar://141462057</a>

Reviewed by Abrar Rahman Protyasha and Megan Gardner.

This is the same test development issue as 287822@main, and as such the commit
message below is copied verbatim.

Since TestWebKitAPI is not a real `UIApplication`, UIKit gets into a state
where its own update cycle is enabled (which is not runtime configurable), but
not started (which is done in `UIApplicationMain()`).

This means that `UIAnimation`-driven animations (like scroll view animations)
will never reach completion, and `scrollViewDidEndScrollingAnimation:` will
never be called.

Fix by leveraging the fact that UIKit&apos;s update cycle is not used for
`UIAnimation`s if `-[UIScreen _isEmbeddedScreen]` is false. This will result in
UIKit using a `CADisplayLink` to drive animations, and ensures completion is
reached without needing to be a `UIApplication`.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAppHighlights.mm:
(TestWebKitAPI::TEST(AppHighlights, AppHighlightCreateAndRestoreAndScroll)):

Canonical link: <a href="https://commits.webkit.org/288848@main">https://commits.webkit.org/288848@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a5e6bd5c19a2d9fcfcfaa734a72f6fe6fe7d6e51

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/84558 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/4283 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/38945 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/89696 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/35607 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/86643 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/4373 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/12254 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/65816 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23645 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/87604 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/3318 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/76856 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/46088 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/3193 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/31067 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/34685 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/74080 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/31863 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/91132 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/11898 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/8680 "Found 1 new test failure: http/tests/security/contentSecurityPolicy/connect-src-star-websocket-allowed.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/74333 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/12125 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72658 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/73458 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/17777 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/16224 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/3337 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13179 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/11850 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/11684 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/15178 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/13430 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
I wonder if we have other embedded API tests failing because of this UIAnimation issue.

pxlcoder committed:
[iOS] TestWebKitAPI.AppHighlights.AppHighlightCreateAndRestoreAndScroll is timing out
https://bugs.webkit.org/show_bug.cgi?id=285882
rdar://141462057

Reviewed by Abrar Rahman Protyasha and Megan Gardner.

This is the same test development issue as 287822@main, and as such the commit
message below is copied verbatim.

Since TestWebKitAPI is not a real `UIApplication`, UIKit gets into a state
where its own update cycle is enabled (which is not runtime configurable), but
not started (which is done in `UIApplicationMain()`).

This means that `UIAnimation`-driven animations (like scroll view animations)
will never reach completion, and `scrollViewDidEndScrollingAnimation:` will
never be called.

Fix by leveraging the fact that UIKit's update cycle is not used for
`UIAnimation`s if `-[UIScreen _isEmbeddedScreen]` is false. This will result in
UIKit using a `CADisplayLink` to drive animations, and ensures completion is
reached without needing to be a `UIApplication`.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAppHighlights.mm:
(TestWebKitAPI::TEST(AppHighlights, AppHighlightCreateAndRestoreAndScroll)):

Canonical link: https://commits.webkit.org/288848@main
